pynucastro/eos/fermi_integrals.py

```diff
@@ -0,0 +1,203 @@
+
+        eta = self.eta
+        beta = self.beta
+        k = self.k
+
+        if first:
+            xsq = x*x
+            xdk = x**(2*k + 1)
+            if ((xsq-eta) < 2.0e2):
+                sqrt_factor = np.sqrt(1 + (xsq*beta/2))
+                exp_factor = np.exp(xsq - eta)
+                denom = exp_factor + 1
+                denomi = 1/denom
+                kernel = 2*xdk*sqrt_factor*denomi
+                denom2i =  1/(4 + 2*beta*xsq)
+
+                fval = kernel
+                df_deta = fval*exp_factor*denomi
+                df_deta2 = (2*exp_factor*denomi - 1)*df_deta
+                df_dbeta = fval*xsq*denom2i
+                df_dbeta2 = -df_dbeta*xsq*denom2i
+                df_deta_dbeta = df_dbeta*exp_factor*denomi
+            else:
+                sqrt_factor = np.sqrt(1 + (xsq*beta/2))
+                denomi = np.exp(eta - xsq)
+                kernel = 2*xdk*sqrt_factor*denomi
+                denom2i =  1/(4 + 2*beta*xsq)
+
+                fval = kernel
+                df_deta = fval
+                df_deta2 = fval
+                df_dbeta = fval*xsq*denom2i
+                df_dbeta2 = -df_dbeta*xsq*denom2i
+                df_deta_dbeta = df_dbeta
+        else:
+            if ((x-eta) < 2.0e2):
+                xdk = x**k
+                sqrt_factor = np.sqrt(1 + (x*beta/2))
+                exp_factor = np.exp(x - eta)
+                denom = exp_factor + 1
+                denomi = 1/denom
+                kernel = xdk*sqrt_factor*denomi
+                denom2i =  1/(4 + 2*beta*x)
+
+                fval = kernel
+                df_deta = fval*exp_factor*denomi
+                df_deta2 = (2*exp_factor*denomi - 1)*df_deta
+                df_dbeta = fval*x*denom2i
+                df_dbeta2 = -df_dbeta*x*denom2i
+                df_deta_dbeta = df_dbeta*exp_factor*denomi
+            else:
+                xdk = x**k
+                sqrt_factor = np.sqrt(1 + (x*beta/2))
+                denomi = np.exp(eta - x)
+                kernel = xdk*sqrt_factor*denomi
+                denom2i =  1/(4 + 2*beta*x)
+
+                fval = kernel
+                df_deta = fval
+                df_deta2 = fval
+                df_dbeta = fval*x*denom2i
+                df_dbeta2 = -df_dbeta*x*denom2i
+                df_deta_dbeta = df_dbeta
+
+        if(eta_der == 0 and beta_der == 0):
+            return fval
+        elif(eta_der == 1 and beta_der == 0):
+            return df_deta
+        elif(eta_der == 2 and beta_der == 0):
+            return df_deta2
+        elif(eta_der == 0 and beta_der == 1):
+            return df_dbeta
+        elif(eta_der == 0 and beta_der == 2):
+            return df_dbeta2
+        elif(eta_der == 1 and beta_der == 1):
+            return df_deta_dbeta
+        else:
+            raise ValueError("Invalid derivative order")
+
+    def _fermi_points(self, eta_der=0):
+
+        eta = self.eta
+
+        D = [3.3609, 4.99551, 3.93830, 4.17444]
```

Comment:
since we only ever need a single eta at a time, I think we should store these parameters in a little more pythonic way, creating a `BreakPoints` class with a1, b1, c1, ... member data, and then a method to evaluate the break points, and then we just store a list or dict of `BreakPoints`.  This will make it more readable.